packages/template-retail-react-app/app/hooks/use-einstein.js

```diff
@@ -440,51 +429,88 @@ const useEinstein = () => {
         return reco
     }
 
+    const getEventUserParameters = async () => {
+        const usid = await getUsidWhenReady()
+        const userParameters = {}
+
+        // If the shopper is authenticated, send the encrypted user ID so recommendations
+        // persist across logins
+        if (isRegistered) {
+            userParameters.userId = await getEncUserIdWhenReady()
+        }
+
+        // Append the `usid` as the `cookieId` value if present. (It should always be present as long
+        // as the user is initilized)
+        if (usid) {
+            userParameters.cookieId = usid
+        } else {
+            console.warn('Missing `cookieId`. For optimal results this value must be defined.')
```

Comment:
under what circumstance does `usid` become falsy? i feel like the else statement is useless？

Reply:
could you explain what do you mean by “optimal results"? I'm under the impression that if `usid` is missing, the feature is broken.

Reply:
In Einstein, there are 2 keys it can use to associate a shopper with an event: the `userId` and the `cookieId`. If both are provided, `userId` takes precedence. 

Technically, without either a `userId` and the `cookieId` nothing is broken in the sense that individual events are still sent to Einstein. Einstein just wouldn't have any idea of who triggered the events. Einstein is degreaded in that if it doesn't know who triggered which event, it is unable to map out the shopper's journey and data like conversion rates become incorrect.

Reply:
Usid can be falsey if someone deletes the usid cookie specifically but not the rest of the auth cookies

Reply:
it seems `usid` never resolves to a falsy value.

Reply:
i would just remove the `if...else...` statement

Reply:
Done. The conditional and warning has been removed